host-controller/src/main/java/org/jboss/as/host/controller/Main.java

```diff
@@ -212,7 +212,8 @@ public static HostControllerEnvironmentWrapper determineEnvironment(String[] arg
         String modulePath = null;
 
         // Note the java.security.manager property shouldn't be set, but we'll check to ensure the security manager should be enabled
-        boolean securityManagerEnabled = System.getSecurityManager() != null || isJavaSecurityManagerConfigured(hostSystemProperties);
+        // [WFCORE-7064] Setting SM is not allowed on JDK24+
+        boolean securityManagerEnabled = Runtime.version().feature() < 24 && (System.getSecurityManager() != null || isJavaSecurityManagerConfigured(hostSystemProperties));
```

Comment:
If this was as follows it would guard/future-proof the System.getSecurityManager() call, while not ignoring the processing of external inputs:

```
boolean securityManagerEnabled = (Runtime.version().feature() < 24 && (System.getSecurityManager() != null) || isJavaSecurityManagerConfigured(hostSystemProperties);
```

Reply:
Fixed.